_posts/2019-06-27-heroku-review-apps-in-multi-tenant-application.markdown

```diff
@@ -0,0 +1,98 @@
+
+### What is Review Apps?
+Review Apps is a feature of Heroku that will automatically deploy each pull request code to a new heroku application. It's an excellent resource to help others (specially code reviewers) to visually see and understand the changes that were made.
+
+<!--more-->
+
+### Enabling Review Apps in Heroku
+The first step is to enable the review apps feature for the desired app using Heroku dashboard.
+
+<div style="text-align: center; width: 500px;">
+  <img src="/blog/assets/images/heroku/enable-review-apps.png">
+</div>
+
+Choose a parent app in your pipeline that’s connected to the correct GitHub repository and click on "**Create an app.json File**".
+
+<div style="text-align: center; width: 500px;">
+  <img src="/blog/assets/images/heroku/enable-review-apps-step-2.png">
+</div>
+
+### The app.json file
+
+After you finish the previous step, a new file called app.json will be committed to the root path of the target repository. This is the config file used by Heroku to create new apps when pull requests are opened. It will look like this:
+
+```
+{
+  "name": "my-new-app",
+  "scripts": {},
+  "env": {
+    "LANG": {
+      "required": true
+    },
+    "RACK_ENV": {
+      "required": true
+    },
+    "DATABASE_URL": {
+      "required": true
+    }
+  },
+  "formation": {},
+  "addons": [
+  ],
+  "buildpacks": [
+  ],
+  "stack": "heroku-18"
+}
+```
+
+As you may guess, the `env` section lists the environment variables that should be copied from the parent app. Take a look in [this link](https://devcenter.heroku.com/articles/github-integration-review-apps#the-app-json-file) to see all available options for the `app.json` file.
+
+### Setup Postgres
+The next step is to add an add-on to connect the apps to heroku postgresql. In order to not affect the data in staging we need to say to heroku to setup the database url in a different environment variable. Remember that the staging database variable was `DATABASE_URL`, so let's use a new one called `REVIEW_APPS_DATABASE` <small>(yes, without the URL part)</small>:
+
+```
+"addons": [
+  {
+    "plan": "heroku-postgresql",
+    "as": "REVIEW_APPS_DATABASE"
+  }
+```
+
+### Copy DB to review app
+
+The recommendation from Heroku is to use the `db/seeds` file to populate your initial database. But for this project we wanted to actually have a copy of the staging database. Since review apps uses a limited resource for database, copy the entire database would cause a database disruption. But since we are using tenants in our application ([apartment gem](https://github.com/influitive/apartment)) our strategy was copy just one schema to the apps and prevent that issue.
```

Comment:
```suggestion
The recommendation from Heroku is to use the `db/seeds` file to populate your initial database. But for this project we wanted to actually have a copy of the staging database. Since review apps uses a limited resource for database, copying the entire database would cause a database disruption. But since we are using tenants in our application ([apartment gem](https://github.com/influitive/apartment)) our strategy was to copy just one schema and prevent that issue.
```